hostpath-provisioner.go

```diff
@@ -177,21 +198,40 @@ func (p *hostPathProvisioner) Delete(volume *v1.PersistentVolume) error {
 }
 
 func calculatePvCapacity(path string) (*resource.Quantity, error) {
-	var stat syscall.Statfs_t
-	err := syscall.Statfs(path, &stat)
+	statfs := &unix.Statfs_t{}
+	err := unix.Statfs(path, statfs)
 	if err != nil {
 		return nil, err
 	}
-	totalSize := stat.Blocks * uint64(stat.Bsize)
-	if totalSize > math.MaxInt64 {
-		glog.Errorf("Calculated total disk size larger than: %d", math.MaxInt64)
-		return nil, errors.New("Total available space doesn't fit in int64")
+
+	// Capacity is total block count * block size
+	capacity := int64(statfs.Blocks) * int64(statfs.Bsize)
+
+	if err != nil {
```

Comment:
There's no new assignment to err from L#202, so do you need this for anything?

Reply:
Nope